tutorials/2d/custom_drawing_in_2d.rst

```diff
@@ -196,333 +209,850 @@ The same applies to the ``draw_rect`` method with ``filled = false``.
+        rotation = 0
+        position = Vector2(60, 60)
+        ...
+
+    func _draw():
+        draw_set_transform(Vector2(-60, -60))
+        ...
 
  .. code-tab:: csharp
 
-    public override void _Draw()
+    public override void _Ready()
     {
-        var center = new Vector2(200, 200);
-        float radius = 80;
-        var color = new Color(1, 0, 0);
+        Rotation = 0;
+        Position = new Vector2(60, 60);
+        ...
+    }
 
-        DrawCircleArc(center, radius, _angleFrom, _angleTo, color);
+    public override void _Draw()
+    {
+        DrawSetTransform(new Vector2(-60.0f, -60.0f));
+        ...
     }
 
+This is the result, rotating around a pivot now on ``(60, 60)``:
 
-Let's run!
-It works, but the arc is rotating insanely fast! What's wrong?
+.. image:: img/draw_godot_rotation.webp
 
-The reason is that your GPU is actually displaying the frames as fast as it can.
-We need to "normalize" the drawing by this speed; to achieve that, we have to make
-use of the ``delta`` parameter of the ``_process()`` function. ``delta`` contains the
-time elapsed between the two last rendered frames. It is generally small
-(about 0.0003 seconds, but this depends on your hardware), so using ``delta`` to
-control your drawing ensures that your program runs at the same speed on
-everybody's hardware.
+If what we wanted to animate was a property inside the ``_draw()`` call, we must remember to
+call ``queue_redraw()`` to force a refresh, as otherwise it would not be updated on screen.
 
-In our case, we simply need to multiply our ``rotation_angle`` variable by ``delta``
-in the ``_process()`` function. This way, our 2 angles will be increased by a much
-smaller value, which directly depends on the rendering speed.
+For example, this is how we can make the robot appear to open and close its mouth, by
+changing the width of its mouth line follow a sinusoidal (:ref:`sin<class_@globalscope_method_sin>`) curve:
 
 .. tabs::
  .. code-tab:: gdscript GDScript
 
-    func _process(delta):
-        angle_from += rotation_angle * delta
-        angle_to += rotation_angle * delta
+    var _mouth_width : float = 4.4
+    var _max_width : float = 7
+    var _t : float = 0
 
-        # We only wrap angles when both of them are bigger than 360.
-        if angle_from > 360 and angle_to > 360:
-            angle_from = wrapf(angle_from, 0, 360)
-            angle_to = wrapf(angle_to, 0, 360)
+    func _process(delta : float):
+        _t += delta
+        _mouth_width = abs(sin(_t) * _max_width)
         queue_redraw()
 
+    func _draw():
+        ...
+        draw_polyline(mouth, white, _mouth_width)
+        ...
+
  .. code-tab:: csharp
 
+    private float _mouthWidth = 4.4f;
+    private float _maxWidth = 7f;
+    private float _t = 0f;
+
     public override void _Process(double delta)
     {
-        _angleFrom += _rotationAngle * (float)delta;
-        _angleTo += _rotationAngle * (float)delta;
+        t += (float)delta;
```

Comment:
```suggestion
        _t += (float)delta;
```

Reply:
Please give me some time before going over it- I will find the same issues while testing, but for example, in this case, I have already changed t to _time as you suggested in my local copy.